src/zenml/utils/metadata_utils.py

```diff
@@ -13,96 +13,118 @@
 #  permissions and limitations under the License.
 """Utility functions to handle metadata for ZenML entities."""
 
-import contextlib
 from typing import Dict, Optional, Union, overload
 from uuid import UUID
 
 from zenml.client import Client
-from zenml.enums import MetadataResourceTypes
+from zenml.enums import MetadataResourceTypes, ModelStages
 from zenml.logger import get_logger
 from zenml.metadata.metadata_types import MetadataType
+from zenml.models import RunMetadataResource
 from zenml.steps.step_context import get_step_context
 
 logger = get_logger(__name__)
 
 
 @overload
-def log_metadata(metadata: Dict[str, MetadataType]) -> None: ...
+def log_metadata(
+    metadata: Dict[str, MetadataType],
+) -> None: ...
 
 
 @overload
 def log_metadata(
     *,
     metadata: Dict[str, MetadataType],
-    artifact_version_id: UUID,
+    step_id: UUID,
 ) -> None: ...
 
 
 @overload
 def log_metadata(
     *,
     metadata: Dict[str, MetadataType],
-    artifact_name: str,
-    artifact_version: Optional[str] = None,
+    step_name: str,
+    run_id_name_or_prefix: Union[UUID, str],
 ) -> None: ...
 
 
 @overload
 def log_metadata(
     *,
     metadata: Dict[str, MetadataType],
-    model_version_id: UUID,
+    run_id_name_or_prefix: Union[UUID, str],
 ) -> None: ...
 
 
 @overload
 def log_metadata(
     *,
     metadata: Dict[str, MetadataType],
-    model_name: str,
-    model_version: str,
+    artifact_version_id: UUID,
 ) -> None: ...
 
 
 @overload
 def log_metadata(
     *,
     metadata: Dict[str, MetadataType],
-    step_id: UUID,
+    artifact_name: str,
+    artifact_version: Optional[str] = None,
 ) -> None: ...
 
 
 @overload
 def log_metadata(
     *,
     metadata: Dict[str, MetadataType],
-    run_id_name_or_prefix: Union[UUID, str],
+    infer_artifact: bool = False,
+    artifact_name: Optional[str] = None,
```

Comment:
Why do those 2 come together here?

```suggestion
    infer_artifact: bool = False,
```

Reply:
You can potentially have a step with multiple outputs, you name one of them through the `artifact_name` and expect the `infer_artifact` to look for it in the step execution.